src/libsystemd/sd-journal/journal-send.c

```diff
@@ -58,7 +58,9 @@ static int journal_fd(void) {
 
         fd_inc_sndbuf(fd, SNDBUF_SIZE);
 
-        if (!__sync_bool_compare_and_swap(&fd_plus_one, 0, fd+1)) {
+        int zero = 0;
+        if (!__atomic_compare_exchange_n(&fd_plus_one, &zero, fd+1,
```

Comment:
Wonder if this is better…
```suggestion
        if (!__atomic_compare_exchange_n(&fd_plus_one, &(int){0}, fd+1,
```

Reply:
I had no idea you could do this.  Learned something new today